app/services/provider_interface/save_condition_statuses.rb

```diff
@@ -0,0 +1,64 @@
+module ProviderInterface
+  class SaveConditionStatuses
+    include ImpersonationAuditHelper
+
+    def initialize(actor:, application_choice:, conditions:)
+      @auth = ProviderAuthorisation.new(actor: actor)
+      @application_choice = application_choice
+      @conditions = conditions
+    end
+
+    def save!
+      auth.assert_can_make_decisions!(application_choice: application_choice, course_option: application_choice.current_course_option)
+
+      audit(auth.actor) do
+        ActiveRecord::Base.transaction do
+          save_conditions_and_update_application!
+        end
+      end
+    end
+
+  private
+
+    attr_reader :auth, :application_choice, :conditions
+
+    def save_conditions_and_update_application!
+      if conditions_met?
+        transition_to_conditions_met!
+      elsif conditions_not_met?
+        transition_to_conditions_not_met!
+      else
+        save_conditions
+      end
+    end
+
+    def conditions_met?
+      conditions.all?(&:met?)
+    end
```

Comment:
This duplicates the logic in the form object, I'll pass in the wizard to this service and use its method